### PR TITLE
Rework the DateRangeFilter component

### PR DESCRIPTION
**Pull request by Windvis**
Instead of the date range component we now use a list of preset ranges with the option to select a custom range if required.

This took _waaay_ longer than intended. Was struggling hard with TS and css fixes for Safari..

### CODE REVIEW

File: app/components/filters/date-range-filter.ts [hunk 1]
```diff
@@ -1,63 +1,189 @@
...
-  }
-
-  @action
-  async dateChange(e: Event, start: string, end: string) {
-    this.changeValue(start, end);
+  /**
+   *
+   * @param params object with {queryParameterName: newValue}
+   */
+  updateQueryParams(params: {
```
Comment (Windvis) [inline app/components/filters/date-range-filter.ts:170]: I copied this from the base class. Using inheritance to share functionality like this is a bad practice and makes things hard to follow (imo). Since it's a simple function call I don't think we need an extra abstraction here.

File: app/components/filters/date-range-filter.ts [hunk 2]
```diff
@@ -1,63 +1,189 @@
...
+}
+
+enum Presets {
+  ThisWeek = 'Deze week',
+  LastWeek = 'Voorbije week',
+  ThisMonth = 'Deze maand',
+  LastMonth = 'Vorige maand',
+  ThisYear = 'Dit jaar',
+  LastYear = 'Vorig jaar',
+  CustomRange = 'Aangepaste periode',
```
Comment (Windvis) [inline app/components/filters/date-range-filter.ts:36]: Still WIP and the specifics are still being worked out. I didn't want to drop everything I had though, since some things can probably be used still.

**Comment by Windvis** [comment]
It seems BNB-290 is caused by changes in this branch. 🕵️

File: app/components/filters/date-range-filter.ts [hunk 3]
```diff
@@ -1,63 +1,242 @@
...
+    return Boolean(this.start) && Boolean(this.end);
+  }
+
+  get hasNoDates(): boolean {
+    return !this.start && !this.end;
+  }
+
+  get isInvalidDateRange(): boolean {
+    return (
+      !!this.start && !!this.end && new Date(this.start) > new Date(this.end)
```
[the remark this answers is not in the document]
Comment (aliokan) [reply app/components/filters/date-range-filter.ts:97]: Typescript and his part of Magic...

**Review by aliokan — APPROVED** [review]
I left some comments, but no blocker. Good work! :+1:

File: app/components/filters/date-range-filter.ts [hunk 4]
```diff
@@ -1,63 +1,242 @@
...
+  get presets(): Preset[] {
+    return [
+      Preset.ThisWeek,
+      Preset.LastWeek,
+      Preset.ThisMonth,
+      Preset.LastMonth,
+      Preset.ThisYear,
+      Preset.LastYear,
+    ];
   }
```
[the remark this answers is not in the document]
Comment (Windvis) [reply app/components/filters/date-range-filter.ts:54]: No good reason it seems. I think I refactored things around.

File: app/components/filters/date-range-filter/date-input.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import Component from '@glimmer/component';
+
+interface Signature {
+  Element: HTMLInputElement;
```
[the remark this answers is not in the document]
Comment (Windvis) [reply app/components/filters/date-range-filter/date-input.ts:4]: This one uses `...attributes` in the template. The other one doesn't (but maybe I forgot, since we're not using Glint yet).

> The Element field declares what type of element(s), if any, the component applies its passed ...attributes to. 

https://typed-ember.gitbook.io/glint/environments/ember/component-signatures

File: app/components/filters/date-range-filter.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Windvis) [reply app/components/filters/date-range-filter.ts:97]: Yea I did some digging and it seems it just hasn't been implemented yet: https://github.com/microsoft/TypeScript/issues/16655

File: app/components/filters/date-range-filter.ts [hunk 6]
```diff
@@ -1,63 +1,242 @@
...
+
+  findPreset(
+    start: ISODateString | null,
+    end: ISODateString | null
+  ): Preset | null {
+    const NO_PRESET = null;
+
+    if (!start && !end) {
+      return NO_PRESET;
+    }
```
[the remark this answers is not in the document]
Comment (Windvis) [reply app/components/filters/date-range-filter.ts:191]: Nope, good catch 👍

File: app/styles/components/_c-interface.scss [hunk 7]
```diff
@@ -13,10 +13,12 @@
   --sidebar: 100%;
 
   flex-basis: var(--sidebar);
+  // Without this `width`, the sidebar dissapears from the page on some breakpoints if the content is too wide
+  width: var(--sidebar);
```
Comment (Windvis) [inline app/styles/components/_c-interface.scss:17]: The issue was already there in master, but it's only reproducible in this PR because of the new select. So I included it here.

Very strange bug, I'm not sure if this is the best fix either, but it shall have to do for now 😄.

**Comment by Windvis** [comment]
Just double checking the labels with the UX team, I deviated a bit from the design there.

File: app/components/filters/filter.ts [hunk 8]
```diff
@@ -7,14 +7,9 @@ interface Signature {
   Args: {
     id: string;
     queryParam: string;
-    queryParamsA: string;
-    queryParamsB: string;
 
     searchField: string;
     options: Array<object>;
-
-    valueA?: string;
```
Comment (aliokan) [inline app/components/filters/filter.ts:16]: Thank you! :pray:

**Review by aliokan — APPROVED** [review]
Looks good, LGTM! :rocket:
